services/provider/server/consumer.go

```diff
@@ -3,85 +3,26 @@ package server
 import (
 	"context"
 	"fmt"
-	"sync"
 
 	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v4/v1alpha1"
 	ifaces "github.com/red-hat-storage/ocs-operator/services/provider/api/v4/interfaces"
 	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
 
-	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
-	clientgocache "k8s.io/client-go/tools/cache"
-	ctrlcache "sigs.k8s.io/controller-runtime/pkg/cache"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
 type ocsConsumerManager struct {
 	client    client.Client
 	namespace string
-	nameByUID map[types.UID]string
-	mutex     sync.RWMutex
 }
 
-func newConsumerManager(ctx context.Context, cl client.Client, namespace string) (*ocsConsumerManager, error) {
-
-	consumerManager := &ocsConsumerManager{
+func newConsumerManager(cl client.Client, namespace string) *ocsConsumerManager {
```

Comment:
The consumer manager now requires a cached client and an indexer for the UID. 
This code need to make sure that the passed client actually support both before allowing usage of the client.

Or at least, be the one to set the index itself